### PR TITLE
Clean up some types

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Cleans up some types. Moves the entity types into the data folder.

Cleaned up according to performance tips from https://github.com/microsoft/TypeScript/wiki/Performance

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/entity/extract_views.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { HassEntities } from "home-assistant-js-websocket";
-import { GroupEntity } from "../../types";
+import { GroupEntity } from "../../data/group";
```
Comment (zsarnett) [inline src/common/entity/extract_views.ts:2]:
```suggestion
import type { GroupEntity } from "../../data/group";
```

File: src/common/dom/apply_themes_on_element.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
+import { Theme } from "../../data/ws-themes";
 import { darkStyles, derivedStyles } from "../../resources/styles";
-import { HomeAssistant, Theme } from "../../types";
+import { HomeAssistant } from "../../types";
```
Comment (zsarnett) [inline src/common/dom/apply_themes_on_element.ts:3]:
```suggestion
import type { HomeAssistant } from "../../types";
```

Comment (zsarnett) [reply src/common/dom/apply_themes_on_element.ts:3]: This type notion is nice to know what are types and what arent. idk if its a performance boost

**Comment by balloob** [comment]
@zsarnett I agree it's nicer, but it's out of scope for this PR :)

**Review by bramkragten — APPROVED** [review]

**Pull request merged**